js/src/secureApi.js

```diff
@@ -136,51 +269,82 @@ export default class SecureApi extends Api {
         this._dappsInterface = dappsInterface;
         this._signerPort = signerPort.toNumber();
       });
-
-    // DEBUG: console.log('SecureApi:connectSuccess', this._transport.token);
-  }
-
-  updateToken (token) {
-    this._transport.updateToken(token.replace(/[^a-zA-Z0-9]/g, ''), false);
-    return this._followConnection();
-    // DEBUG: console.log('SecureApi:updateToken', this._transport.token, connectState);
   }
 
-  get dappsPort () {
-    return this._dappsPort;
+  /**
+   * Try to generate an Authorization Token.
+   * Then try to connect with the new token.
+   */
+  _generateAuthorizationToken () {
+    return this.signer
+      .generateAuthorizationToken()
+      .then((token) => this._connectWithToken(token));
   }
 
-  get dappsUrl () {
-    let hostname;
+  /**
+   * Get the next token to try, if any left
+   */
+  _getNextToken () {
+    // Get the first not-tried token
+    const nextTokenIndex = this._tokens.findIndex((t) => !t.tried);
 
-    if (window.location.hostname === 'home.parity') {
-      hostname = 'dapps.parity';
-    } else if (!this._dappsInterface || this._dappsInterface === '0.0.0.0') {
-      hostname = window.location.hostname;
-    } else {
-      hostname = this._dappsInterface;
+    // If no more tokens to try, user has to enter a new one
+    if (nextTokenIndex < 0) {
+      return null;
     }
 
-    return `http://${hostname}:${this._dappsPort}`;
+    const nextToken = this._tokens[nextTokenIndex];
+    return nextToken;
   }
 
-  get signerPort () {
-    return this._signerPort;
+  _resetTokens () {
+    this._tokens = this._tokens.map((token) => ({
+      ...token,
+      tried: false
+    }));
   }
 
-  get isConnecting () {
-    return this._isConnecting;
+  _sanitiseToken (token) {
+    return token.replace(/[^a-zA-Z0-9]/g, '');
   }
 
-  get isConnected () {
-    return this._transport.isConnected;
+  _saveToken (token) {
+    window.localStorage.setItem('sysuiToken', token);
   }
 
-  get needsToken () {
-    return this._needsToken;
-  }
+  /**
+   * Promise gets resolved when the node is up
+   * and running (it might take some time before
+   * the node is actually ready even when the client
+   * is connected).
+   *
+   * We check that the `parity_enode` RPC calls
+   * returns successfully
+   */
+  _waitUntilNodeReady () {
+    return this
+      .parity.enode()
```

Comment:
It is a tough one, since from the UI perspective, everything works even without the enode. Well, it appears working. I suppose my preference would have been an "enode not ready" status warning, instead of waiting for it - not to be done here and now though, that check just has me slightly uncomfortable...